### PR TITLE
2FA Account flow improvements, AccountBlocked screen

### PR DESCRIPTION
**Pull request by kenanmamedoff**

### CODE REVIEW

File: brd-ios/breadwallet/Helpers/Networking/NetworkingError.swift [hunk 1]
```diff
@@ -44,6 +48,11 @@ enum NetworkingError: FEError {
         case .serverAtCapacity:
             return L10n.ErrorMessages.somethingWentWrong
             
+        case .twoStepInvalid:
+            return """
+Invalid code. You have 3 more attempts, after that your RockWallet account will be blocked. You’ll still be able to access your self-custodial features.
```
Comment (dgacevic) [inline brd-ios/breadwallet/Helpers/Networking/NetworkingError.swift:53]: Should be localized

File: brd-ios/breadwallet/src/Redesign/Models/ServerResponse.swift [hunk 2]
```diff
@@ -14,6 +14,11 @@ struct ServerResponse: Decodable {
     enum ErrorType: String, Decodable {
         case exchangesUnavailable = "Exchanges unavailable"
         case biometricAuthentication = "Biometric authentication"
+        case twoStepRequired = "Required 2FA"
+        case twoStepInvalid = "Invalid 2FA code"
+        case twoStepInvalidRetryable = "Invalid code"
+        case twoStepBlockedAccount = "Account blocked"
+        case twoStepInvalidCodeBlockedAccount = "Invalid 2FA code, account blocked"
```
Comment (dgacevic) [inline brd-ios/breadwallet/src/Redesign/Models/ServerResponse.swift:21]: Should these be localized?

File: brd-ios/breadwallet/src/Redesign/Scenes/RegistrationConfirmation/RegistrationConfirmationPresenter.swift [hunk 3]
```diff
@@ -46,16 +55,33 @@ final class RegistrationConfirmationPresenter: NSObject, Presenter, Registration
             instructions = "\(L10n.AccountCreation.enterCode)\(email)"
             
         case .twoStepApp, .twoStepAppLogin:
-            title = "Enter the code from your Authenticator app"
+            title = "Enter one of your backup codes"
```
Comment (dgacevic) [inline brd-ios/breadwallet/src/Redesign/Scenes/RegistrationConfirmation/RegistrationConfirmationPresenter.swift:58]: 👀

File: brd-ios/breadwallet/src/Redesign/Scenes/RegistrationConfirmation/RegistrationConfirmationPresenter.swift [hunk 4]
```diff
@@ -46,16 +55,33 @@ final class RegistrationConfirmationPresenter: NSObject, Presenter, Registration
...
             instructions = ""
+        
+        case .enterAppBackupCode:
+            title = "Enter the code from your Authenticator app"
+            instructions = "Confirm you’ve stored your backup codes securely by entering one of them."
```
Comment (dgacevic) [inline brd-ios/breadwallet/src/Redesign/Scenes/RegistrationConfirmation/RegistrationConfirmationPresenter.swift:63]: .

File: brd-ios/breadwallet/src/Redesign/Scenes/RegistrationConfirmation/RegistrationConfirmationPresenter.swift [hunk 5]
```diff
@@ -46,16 +55,33 @@ final class RegistrationConfirmationPresenter: NSObject, Presenter, Registration
...
         
         if UserManager.shared.profile?.status == .emailPending {
-            help.append(ButtonViewModel(title: L10n.AccountCreation.changeEmail, isUnderlined: true))
+            help.append(ButtonViewModel(title: L10n.AccountCreation.changeEmail,
+                                        isUnderlined: true,
+                                        callback: viewController?.changeEmailTapped))
+        }
+        
+        if confirmationType == .twoStepAppLogin {
+            help = [ButtonViewModel(title: "I can’t access my Authenticator App",
```
Comment (dgacevic) [inline brd-ios/breadwallet/src/Redesign/Scenes/RegistrationConfirmation/RegistrationConfirmationPresenter.swift:78]: .

File: brd-ios/breadwallet/src/Redesign/Scenes/Swap/Validation/AccountBlockedViewController.swift [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    override var imageName: String? { return Asset.unlockWalletDisabled.name }
+    override var titleText: String? { return "Your RockWallet account is currently disabled." }
+    override var descriptionText: String? { return """
+It appears that you're unable to verify your account. For security reasons, we have temporarily disabled it.
+
+Don't worry, your data is safe, and you can still access your self-custodial funds from your homepage. For more information, please contact support.
+""" }
+    override var buttonViewModels: [ButtonViewModel] {
+        return [
+            .init(title: "BACK TO HOME PAGE", callback: { [weak self] in
```
Comment (dgacevic) [inline brd-ios/breadwallet/src/Redesign/Scenes/Swap/Validation/AccountBlockedViewController.swift:27]: .

File: brd-ios/breadwallet/src/Redesign/Scenes/Swap/Validation/AccountBlockedViewController.swift [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+                self?.didTapMainButton?()
+            }),
+            .init(title: "Contact support", isUnderlined: true, callback: { [weak self] in
```
Comment (dgacevic) [inline brd-ios/breadwallet/src/Redesign/Scenes/Swap/Validation/AccountBlockedViewController.swift:30]: .

File: brd-ios/breadwallet/src/Redesign/Scenes/Swap/Validation/AccountBlockedViewController.swift [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+
+import UIKit
+
+extension Scenes {
+    static let AccountBlocked = AccountBlockedViewController.self
+}
+
+class AccountBlockedViewController: BaseInfoViewController {
+    override var imageName: String? { return Asset.unlockWalletDisabled.name }
+    override var titleText: String? { return "Your RockWallet account is currently disabled." }
```
Comment (dgacevic) [inline brd-ios/breadwallet/src/Redesign/Scenes/Swap/Validation/AccountBlockedViewController.swift:19]: .

File: brd-ios/breadwallet/src/Redesign/Scenes/Swap/Validation/AccountBlockedViewController.swift [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    override var descriptionText: String? { return """
+It appears that you're unable to verify your account. For security reasons, we have temporarily disabled it.
```
Comment (dgacevic) [inline brd-ios/breadwallet/src/Redesign/Scenes/Swap/Validation/AccountBlockedViewController.swift:21]: .

File: brd-ios/breadwallet/src/Redesign/Models/ServerResponse.swift (hunk 2 above)
Comment (kenanmamedoff) [reply brd-ios/breadwallet/src/Redesign/Models/ServerResponse.swift:21]: no. these are error types and will be updated. not going to be visible to the users.

**Review by dgacevic — APPROVED** [review]

**Pull request merged**